Sources/SourceKitLSP/Swift/Diagnostic.swift

```diff
@@ -71,10 +59,46 @@ extension CodeAction {
   }
 
   init?(_ fixIt: FixIt, in snapshot: DocumentSnapshot) {
-    // FIXME: Once https://github.com/apple/swift-syntax/pull/2226 is merged and
-    // FixItApplier is public, use it to compute the edits that should be
-    // applied to the source.
-    return nil
+    var textEdits = [TextEdit]()
+    for edit in fixIt.edits {
+      guard let startPosition = snapshot.position(of:edit.range.lowerBound),
+        let endPosition = snapshot.position(of:edit.range.upperBound) else {
+        continue
+      }
+      textEdits.append(TextEdit(range: startPosition..<endPosition, newText: edit.replacement))
+    }
+
+    guard let generatedTitle = Self.title(for: textEdits, in: snapshot) else {
+      return nil
+    }
```

Comment:
Fix-Its from swift-syntax always have a message (`fixIt.message.message` I think) and I think we should be using that instead of synthesizing a title here. I think the extraction of `title(for:in:)` makes sense regardless though.

Reply:
i tried to use `fixIt.message.message` instead of `title(for:in:)`, but it failed the test, so we might not be able to do it. 
```
Expected: 
"Replace 'Multi ' with 'Multiident '..."
"Replace 'Multi ' with 'MultiIdent '..."

Actual:
"join the identifiers together",
"join the identifiers together with camel-case"
```

Reply:
Can’t you just update the test case to expect `join the identifiers together`?

Also, it would be good to uppercase the first character of the diagnostic, similar to what I did here: https://github.com/apple/sourcekit-lsp/pull/784

Reply:
you're right, i put the code `protocol Multi ident {}` in a swift playground.
the error message was 
```
Join the identifiers together
Join the identifiers together with camel-case
```
let me correct it.